context-propagation/src/main/java/io/micrometer/context/ContextSnapshot.java

```diff
@@ -127,17 +128,41 @@ default Executor wrapExecutor(Executor executor) {
      * snapshot around the invocation of any executed task.
      * @param executorService the executorService to instrument
      */
-    default ExecutorService wrapExecutorService(ExecutorService executorService) {
-        return new ContextPropagatingExecutorService<>(executorService, this);
+    static ExecutorService wrapExecutorService(ExecutorService executorService) {
+        return new ContextPropagatingExecutorService<>(executorService);
+    }
+
+    /**
+     * Return a new {@code ExecutorService} that sets {@code ThreadLocal} values from the
+     * snapshot around the invocation of any executed task.
+     * @param executorService the executorService to instrument
+     * @param contextSnapshot supplier of the {@link ContextSnapshot} - instruction on who
+     * to retrieve {@link ContextSnapshot} when tasks are scheduled
+     */
+    static ExecutorService wrapExecutorService(ExecutorService executorService,
+            Supplier<ContextSnapshot> contextSnapshot) {
+        return new ContextPropagatingExecutorService<>(executorService, contextSnapshot);
+    }
```

Comment:
Now that these are no longer instance methods, let's move them down below, after the other static methods.